src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
The `die` method uses `indexOf` and `splice` to remove the dead animal from the `Animal.alive` array. This can be inefficient if the array grows large. Consider using `filter` to create a new array excluding the dead animal, which avoids the nested loop caused by `indexOf` and `splice`.